packages/ui-expandable/src/Expandable/index.tsx

```diff
@@ -26,69 +26,112 @@ import PropTypes from 'prop-types'
 import { controllable } from '@instructure/ui-prop-types'
 import { uid } from '@instructure/uid'
 import { createChainedFunction } from '@instructure/ui-utils'
-const toggleExpanded = ({ expanded }: any) => ({ expanded: !expanded })
+
+const toggleExpanded = ({ expanded }: { expanded: boolean }) => ({
+  expanded: !expanded
+})
+
+// TODO: try to type it so that the return object contains the prop object
```

Comment:
It was a nightmare to type, so in the end I just added a TODO. If you have a tip how to do it without making everything Generic, let me know please :)

Reply:
You could do something like:
```
type GetToggleProps = <P>(props: P & { onClick?: (...args: any[]) => any }) => {
  'aria-controls': string
  'aria-expanded': boolean
  onClick?: (...args: any[]) => any
} & P
```
[playground](https://www.typescriptlang.org/play?noUnusedParameters=true&target=3&inlineSourceMap=false&inlineSources=false&isolatedModules=false&noFallthroughCasesInSwitch=true&ts=4.3.5#code/C4TwDgpgBA4hwBUD2BzFAbCAFATksAzlALxQA8WAfABRh6EBcUWUAZFAN5RIB2AwugCWAYwDWAfibUAdLICGOFASZyeIANoBdAJQlKUVSCgBfXcX0cAUFCgByBYLkBaYb2B50BW0wLvBPFGs7B2cIAA8wVQATCCjvKAAjJCRMVSDeAREJKVlpBSUVNS0zfUNLYzZmS0tXHl8oYAhfJjhEVAxseiJSWi6STgq5IkMSziCceABXHB4xmxtcunwCABogm3scRxc3Dy8mWwBGQ4AmWzX54K3QiOjY+IAzOU8IC5Ny6sbfag4hqIYACymPIEKKWL7AH5-QHAv4PBJRIA)

Reply:
note that you will need a `as any` cast here: 
`getToggleProps: (props = {} as any) => {` because the default parameter would determine the generic type otherwise.

Reply:
Also onClick should have this signature: ` onClick?: (event: Event) => void`

Reply:
https://stackoverflow.com/questions/54225839/required-vs-inferred-generic-types-in-typescript

Reply:
Thanks!